pkg/lib/indexconfig/configdefs.go

```diff
@@ -0,0 +1,111 @@
+package indexconfig
```

Comment:
One other observation I have is regarding the word "config" everywhere -- I know we call it the declarative index config, but I'm wondering what value the word config provides. Go packages are all namespaced scoped. I'm wondering if we had something like `pkg/lib/declarative/...` instead and then remove the word config from the names altogether.

Reply:
I'm not sure `declarative` by itself conveys anything honestly. I've tried to convey the message in such a way that someone who's read the enhancement once can read the code easily. The word declarative isn't used more than 2-3 times in the enhancement itself too btw.

Reply:
Before I saw this PR, I had named a package for a similar purpose `declcfg`. I'm good with `declarative` too but figured I'd throw out what my first instinct was.